### PR TITLE
Updates to EditCavcRemandForm and Jest Tests

### PR DESCRIPTION
**Pull request by jcq**
Connects [CASEFLOW-1052](https://vajira.max.gov/browse/CASEFLOW-1052)

### Description
This is a follow-up to earlier work on 1052 that adjusts the logic for handling the fields for `decisionType` and `remandType` based on the mode (`add` or `edit`) of the component.

This also adds back in Jest tests for `EditCavcRemandForm` that were removed in order to get earlier changes merged.

### Acceptance Criteria
- [x] Code compiles correctly
- [x] `EditCavcRemandForm` works as it did before in edit mode
- [x] Jest tests for `EditCavcRemandForm` are present

### Testing Plan
1. View the "editing" stories in Storybook, ensure that forms work as they did prior
2. View in Storybook, ensure the default ("add") mode works correctly
3. Verify that editing a CAVC remand with this branch behaves the same way it does in `master`

### User Facing Changes
Should look the same

### CODE REVIEW

File: client/app/queue/cavc/EditCavcRemandForm.jsx [hunk 1]
```diff
@@ -215,40 +222,57 @@ export const EditCavcRemandForm = ({
...
+            strongLabel
+            vertical
           />
-          <TextField
-            inputRef={register}
-            name="remandType"
-          />
-        </div>
+        ) : (
+          <>
```
Comment (n1zyy) [inline client/app/queue/cavc/EditCavcRemandForm.jsx:238]: What is the meaning of the `<>` tags here?

**Review by n1zyy — COMMENTED** [review]
I left a React n00b question inline.

I don't see anything that looks problematic to me here, but I will let Paula take the formal review.

File: client/app/queue/cavc/EditCavcRemandForm.jsx (hunk 1 above)
Comment (jcq) [reply client/app/queue/cavc/EditCavcRemandForm.jsx:238]: That is a shorthand version of `<React.Fragment>`

See docs:
https://reactjs.org/docs/fragments.html

**Review by pbradin — APPROVED** [review]
Looks good

**Pull request merged**